7 Web Development - User Service View/userServiceView-ux/src/rc/UserCardWidget.js

```diff
@@ -0,0 +1,59 @@
+/** @jsx spandrel.jsx */
+/* eslint-env browser */
+define([
+	"baja!",
+	"bajaux/spandrel",
+	'bajaux/mixin/subscriberMixIn',
+	"css!nmodule/userServiceView/rc/UserCard"
+], function(
+	baja,
+	spandrel,
+	subscriberMixIn,
+) {
+	"use strict";
+
+	return class UserCardWidget extends spandrel((user) => {
+
+		function extractInitials(name) {
+			return name.split(" ")
+				.map(token => token.length > 0 ? token[0] : "")
+				.join("")
+				.toUpperCase();
+		}
+
+		const roles = user.getRoles().split(",")
+			.filter(role => role !== "")
+			.map(baja.SlotPath.unescape);
+		const username = baja.SlotPath.unescape(user.getName().trim());
+		const fullname = user.getFullName().trim();
+		const initials = extractInitials(fullname) || extractInitials(username);
```

Comment:
Technically I think this deviates slightly from the behavior demonstrated in the screenshot. I guess I didn't spell this out in the description, but the idea was that if a `fullName` is defined, you'd use the first letters from each word in the `fullName` (like you're doing here). But if only a `name` is defined, you'd use the first two letters so that generally someone's initials are always two letters long.

Nothing wrong with this - I'm just pointing it out.